File: server/app/models/db/general_resource/language_response.py
from pydantic import BaseModel
from typing import List
from app.db.enums import AvailableLanguage
from app.models.db.general_resource.dialect_response import DialectResponse
from app.models.db.general_resource.course_response import CourseResponse


class LanguageResponse(BaseModel):
    id: int
    language: AvailableLanguage
    image: str
    text_color: str
    dialects: List[DialectResponse]
    courses: List[CourseResponse]

File: client/types/response_models/LanguageResponse.ts
import { AvailableLanguage } from "../enums";
import { DialectResponse } from "./DialectResponse";
import { CourseResponse } from "./CourseResponse";

export interface LanguageResponse {
    id: number;
    language: AvailableLanguage;
    image: string;
    text_color: string;
    dialects: DialectResponse[];
    courses: CourseResponse[];
}
